app/controllers/concerns/signature_verification.rb

```diff
@@ -83,14 +83,21 @@ def signed_request_account
     raise SignatureVerificationError, "Public key not found for key #{signature_params['keyId']}" if account.nil?
 
     signature             = Base64.decode64(signature_params['signature'])
-    compare_signed_string = build_signed_string
+    compare_signed_string = build_signed_string(request_target_quirk: true)
```

Comment:
Is the quirk code really required? What would break without it?

Reply:
At least GoToSocial and Friendica make signed queries with query strings. In GoToSocial's case, it's to fetch replies, in Friendica's case, it's to fetch the inbox. Changing this without keeping the quirk means there is no backward compatibility for them when querying instances with `AUTHORIZED_FETCH` on.

I have seen no occurrence yet of Mastodon itself querying an URL with a query string (but I'm only observing a small single-user instance which may only exhibit a small set of interactions with just a fraction of the implementations out there).